### PR TITLE
[ticket/15163] Escape curly braces in smilies HTML attributes

### PR DESCRIPTION
**Pull request by JoshyPHP**
PHPBB3-15163

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15163

### CODE REVIEW

**Comment by JoshyPHP** [comment]
As far as I can tell, tests pass with the exception of the build that requires npm, which fails for unrelated reasons.

File: phpBB/phpbb/textformatter/s9e/factory.php [hunk 1]
```diff
@@ -442,6 +442,20 @@ protected function configure_autolink(Configurator $configurator)
...
+	*
+	* Escapes "HTML special chars" for obvious reasons and curly braces to avoid them
+	* being interpreted as an attribute value template
+	*
+	* @param  string $value Original string
+	* @return string        Escaped string
+	*/
+	protected function escape_html_attribute($value)
+	{
+		return htmlspecialchars(strtr($value, ['{' => '{{', '}' => '}}']));
```
Comment (Nicofuma) [inline phpBB/phpbb/textformatter/s9e/factory.php]: We could use `twig_escape_filter()` which implements a stronger escaper but I'm not sure it's a good idea....

Comment (JoshyPHP) [inline phpBB/phpbb/textformatter/s9e/factory.php]: That function escapes literals to be used in XSL templates, I don't think Twig follows exactly the same syntax. I'm going to update this line to explicit specify it escapes for XML.

File: phpBB/phpbb/textformatter/s9e/factory.php [hunk 2]
```diff
@@ -442,6 +442,20 @@ protected function configure_autolink(Configurator $configurator)
 	}
 
 	/**
+	* Escape a string to be used in an HTML attribute
```
Comment (Nicofuma) [inline phpBB/phpbb/textformatter/s9e/factory.php]: please specify it is for an xsl template

Comment (JoshyPHP) [inline phpBB/phpbb/textformatter/s9e/factory.php]: Done.

**Comment by Nicofuma** [comment]
Merged in 3.2.x by e0ae49d

**Pull request closed without merging**